core/runtime/bindings.rs

```diff
@@ -368,6 +370,34 @@ pub extern "C" fn host_initialize_import_meta_object_callback(
   let resolve_key =
     v8::String::new_external_onebyte_static(scope, b"resolve").unwrap();
   meta.set(scope, resolve_key.into(), val.into());
+
+  // For `file:` URL we provide additional `filename` and `dirname` values
+  let Ok(name_url) = Url::parse(&name) else {
+    return;
```

Comment:
Nitpick: Using returns like this is great when the function does one thing, but this function does a lot (it can lead to bugs if someone goes and adds something to the bottom of this function that should still work when this returns). Maybe we should extract the code below to a new function or perhaps we should just nest the if statements?

Reply:
Done.